cou/utils/openstack_lookup.csv

```diff
@@ -1,24 +1,24 @@
-service,ussuri-lower_version,ussuri-upper_version,victoria-lower_version,victoria-upper_version,wallaby-lower_version,wallaby-upper_version,xena-lower_version,xena-upper_version,yoga-lower_version,yoga-upper_version
-aodh,10.0.0,11.0.0,11.0.0,12.0.0,12.0.0,13.0.0,13.0.0,14.0.0,14.0.0,15.0.0
-barbican,10.0.0,11.0.0,11.0.0,12.0.0,12.0.0,13.0.0,13.0.0,14.0.0,14.0.0,15.0.0
-ceilometer,14.0.0,15.0.0,15.0.0,16.0.0,16.0.0,17.0.0,17.0.0,18.0.0,18.0.0,19.0.0
-cinder,16.0.0,17.0.0,17.0.0,18.0.0,18.0.0,19.0.0,19.0.0,20.0.0,20.0.0,21.0.0
-designate,10.0.0,11.0.0,11.0.0,12.0.0,12.0.0,13.0.0,13.0.0,14.0.0,14.0.0,15.0.0
-glance,20.0.0,21.0.0,21.0.0,22.0.0,22.0.0,23.0.0,23.0.0,24.0.0,24.0.0,25.0.0
-heat,14.0.0,15.0.0,15.0.0,16.0.0,16.0.0,17.0.0,17.0.0,18.0.0,18.0.0,19.0.0
-horizon,17.0.0,18.4.0,18.4.0,19.0.0,19.0.0,20.0.0,20.0.0,20.2.0,20.2.0,22.2.0
-keystone,17.0.0,18.0.0,18.0.0,19.0.0,19.0.0,20.0.0,20.0.0,21.0.0,21.0.0,22.0.0
-manila,10.0.0,11.0.0,11.0.0,12.0.0,12.0.0,13.0.0,13.0.0,14.0.0,14.0.0,15.0.0
-masakari,9.0.0,10.0.0,10.0.0,11.0.0,11.0.0,12.0.0,12.0.0,13.0.0,13.0.0,14.0.0
-neutron,16.0.0,17.0.0,17.0.0,18.0.0,18.0.0,19.0.0,19.0.0,20.0.0,20.0.0,21.0.0
-nova,21.0.0,22.0.0,22.0.0,23.0.0,23.0.0,24.0.0,24.0.0,25.0.0,25.0.0,26.0.0
-octavia,6.0.0,7.0.0,7.0.0,8.0.0,8.0.0,9.0.0,9.0.0,10.0.0,10.0.0,11.0.0
-placement,3.0.0,4.0.0,4.0.0,5.0.0,5.0.0,6.0.0,6.0.0,7.0.0,7.0.0,8.0.0
-swift,2.24.0,2.26.0,2.26.0,2.27.0,2.27.0,2.28.0,2.28.0,2.29.0,2.29.0,2.30.0
-ceph,15.2.0,16.2.0,15.2.0,16.2.0,16.2.0,17.2.0,16.2.0,17.2.0,17.2.0,18.2.0
-gnocchi,4.3.0,4.4.0,4.4.0,4.4.1,4.4.0,4.4.1,4.4.1,4.4.2,4.4.1,4.4.2
-designate-bind,9.16.1,9.18.12,9.16.1,9.18.12,9.16.1,9.18.12,9.16.1,9.18.12,9.16.1,9.18.12
-ovn,20.03,22.09,20.03,22.09,20.12,22.09,21.09,22.09,22.03,22.09
-mysql,8,9,8,9,8,9,8,9,8,9
-rabbitmq-server,3.8,3.9,3.8,3.9,3.8,3.9,3.8,3.9,3.8,3.10
-vault,1.7,1.8,1.7,1.8,1.7,1.8,1.7,1.8,1.7,1.9
+service,ussuri-lower_version,ussuri-upper_version,victoria-lower_version,victoria-upper_version,wallaby-lower_version,wallaby-upper_version,xena-lower_version,xena-upper_version,yoga-lower_version,yoga-upper_version,zed-lower_version,zed-upper_version,antelope-lower_version,antelope-upper_version,bobcat-lower_version,bobcat-upper_version,caracal-lower_version,caracal-upper_version

```

Comment:
We don't know `caracal-upper_version` because next release is not known, and so we don't know next incompatible version. Here, I simply increase the version (major / minor / bug) by one

Reply:
I think it's a valid assumption for packages like nova, keystone and etc. I would do the same. However, there are some packages that we should keep an eye on it like `horizon` that it's not always that increases by one.

Now for auxiliary charms like ceph, ovn, rabbitmq we need to ask for the openstack-team. I'm not completely sure that will increase for caracal

Reply:
An idea would be always increase the `PATCH` by one at the unknown next incompatible version so that it effectively defined a lower bound: e.g. 1.7 (lower) and 1.7.1 (upper)

Reply:
I am ok with this solution. But this rule need to be in the document.

Reply:
it's in the document: https://github.com/canonical/charmed-openstack-upgrader/pull/449/files#diff-bde25d5c5361a81248b18ca63a3d984f41d7e2fa37c45eb00bab218c57573be9R37-R40

Reply:
Can you open an issue to properly limit caracal versions when possible? Just to not lose track

Reply:
Created: https://github.com/canonical/charmed-openstack-upgrader/issues/456